### DISCUSSION TITLE
How can I make multiline text in a table cell?

### DISCUSSION THREAD

**Discussion by butonic** [speedata/boxesandglue#6] [General]
I am considering moving a legacy PHP tool where I use mpdf to golang  It lets me edit a mail template as HTML and then I can turn it into a nice PDF. bagme seems to be the right fit in the golang world, because I don't want to run a headless chrome to render the PDF. I am really impressed in what you built already. So, keep it up! Awesome work!

Now, I can use a table to split an area into two colums like below. But the `<p>` and `<br/>` elements in the table cells have no effect. The `<br/>` elements below the table work just fine.

```html
<table style="width:100%; border-spacing:0; ">
<tbody>
<tr>
	<td valign="top"><!-- Anschrift -->
        {{.Address}}
	</td>

	<td style="text-align: right;"><!-- Absender -->
	<p>Michael Mustermann</p>
	<p>Musterstraße 51</p>
	<br/>
	12345 Musterstadt<br/>
	<br/>
	{{if eq .Plattform "dawanda"}}
        E-Mail: dawanda@mustershop.example<br/>
	{{else if eq .Plattform "etsy"}}
        E-Mail: etsy@mustershop.example<br/>
	{{else}}
        E-Mail: shop@mustershop.example<br/>
    {{end}}
	<br/>
	Steuernummer: 12/345/67890<br/>
	<br/>
	Datum: {{.Datum}}
	</td>
</tr>
</tbody>
</table>
Rechnung Nr. {{.Rechnungsjahr}}-{{.Rechnungsnummer}}<br/>
<br/>
<br/>
<br/>
Sehr geehrte Kundin, sehr geehrter Kunde,<br/>
<br/>
für Ihre Bestellung{{if .Bestellnummer}}
<!-- Bestellnummer nur bei Dawanda anzeigen, extra Bestellnummerfeld-->
 mit der Dawanda-Bestellnummer {{.Bestellnummer}}
{{end}} bedanke ich mich und berechne für meine Leistungen:<br/>
<br/>
```

I know I can change the HTML to do what I want, but typically online HTML editors will produce this kind of simple markup. And I want the generated HTML to be editable by my end user.

 So, is that a bug or should I use a different way to produce newlines in the table cell?

I know you are using latex like layouting, but I still need newlines ;-)

**Comment by pgundlach** [comment]
This is a bug. Thank you very much for posting, I'll take a look.

**Comment by pgundlach** [comment]
Just a short feedback: this is more complicated than I initially thought (a conceptual error, needs rethinking on some part of the code), so I can't give you an ETA. But again, thank you for posting.

**Reply by butonic** [reply]
no worries, glad I could point out an interesting bug ;-)
